forge-game/src/main/java/forge/game/GameAction.java

```diff
@@ -916,7 +916,7 @@ public final Card moveToVariantDeck(Card c, ZoneType zone, int deckPosition, Spe
     }
 
     public final CardCollection exile(final CardCollection cards, SpellAbility cause, Map<AbilityKey, Object> params) {
-        CardZoneTable table = new CardZoneTable();
+        CardZoneTable table = new CardZoneTable(game.getLastStateBattlefield(), game.getLastStateGraveyard());
```

Comment:
it seems to only affect the exile method, but shouldn't this also try to use the LastState from Cause if able?

or try to get them from params?